cmd/upgrade.go

```diff
@@ -0,0 +1,135 @@
+package cmd
+
+import (
+	"github.com/Azure/acs-engine/pkg/acsengine"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	log "github.com/Sirupsen/logrus"
+	"github.com/satori/go.uuid"
+	"github.com/spf13/cobra"
+)
+
+const (
+	upgradeName             = "upgrade"
+	upgradeShortDescription = "upgrades an existing Kubernetes cluster"
+	upgradeLongDescription  = "upgrades an existing Kubernetes cluster, first replacing masters, then nodes"
+)
+
+type upgradeCmd struct {
+	// user input
+	authMethod          string
+	clientSecret        string
+	resourceGroupName   string
+	kubeconfigPath      string
+	rawClientID         string
+	rawSubscriptionID   string
+	rawAzureEnvironment string
+
+	// parsed
+	clientID         uuid.UUID
+	subscriptionID   uuid.UUID
+	azureEnvironment azure.Environment
+
+	// derived
+	tenantID              string
+	servicePrincipalToken *adal.ServicePrincipalToken
+}
+
+func NewUpgradeCmd() *cobra.Command {
+	uc := upgradeCmd{}
+
+	upgradeCmd := &cobra.Command{
+		Use:   upgradeName,
+		Short: upgradeShortDescription,
+		Long:  upgradeLongDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return uc.run(cmd, args)
+		},
+	}
+
+	f := upgradeCmd.Flags()
```

Comment:
Doesn't it need to take the APIModel or directory to the generated template folder that has the APIModel file?